src/promo/entity/promo.entity.ts

```diff
@@ -0,0 +1,39 @@
+import type { IPromoCode } from 'promo/promo.interface';
+import { PromocodeState } from 'promo/promocode.enum';
+import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';
+
+@Entity({ name: 'promocode' })
+export class PromocodeEntity implements IPromoCode {
+  @PrimaryGeneratedColumn()
+    id: string;
+
+  @Column()
+    name: string;
+
+  @Column()
+    createdAt: string;
+
+  @Column()
+    percent: number;
+
+  @Column()
+    currentState: PromocodeState;
```

Comment:
don't think that we need this field. promocode is active if it's not deleted. we can add check that deletedAt field is null to get al active promocodes.